File: sic-api/Entities/Su/SuBusiness.cs
using sic_api.Attributes;
using sic_api.Entities.Db;
using sic_api.Model.Storage;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace sic_api.Entities.Su;

[Table("su_business")]
public class SuBusiness : BaseEntity
{

    [MaxLength(30)]
    [Column("tax_id")]
    public string? TaxId { get; set; }

    [Required]
    [MaxLength(30)]
    [Column("business_code")]
    public string BusinessCode { get; set; } = default!;

    [MaxLength(30)]
    [Column("branch_code")]
    public string? BranchCode { get; set; }

    [Required]
    [MaxLength(100)]
    [Column("person_type")]
    public string PersonType { get; set; } = default!;

    [Required]
    [ForeignKey(nameof(Title))]
    [Column("title_id")]
    public Guid TitleId { get; set; }
    public DbTitle Title { get; set; } = default!;

    [Required]
    [MaxLength(100)]
    [Column("first_name_en")]
    public string FirstNameEn { get; set; } = default!;

    [MaxLength(100)]
    [Column("middle_name_en")]
    public string? MiddleNameEn { get; set; }

    [MaxLength(100)]
    [Column("last_name_en")]
    public string? LastNameEn { get; set; }

    [Required]
    [MaxLength(100)]
    [Column("first_name_local")]
    public string FirstNameLocal { get; set; } = default!;

    [MaxLength(100)]
    [Column("middle_name_local")]
    public string? MiddleNameLocal { get; set; }

    [MaxLength(100)]
    [Column("last_name_local")]
    public string? LastNameLocal { get; set; }

    [Required]
    [ForeignKey(nameof(Country))]
    [Column("country_id")]
    public Guid CountryId { get; set; }
    public DbCountry Country { get; set; } = default!;

    [Required]
    [Column("support_local_address")]
    public bool SupportLocalAddress { get; set; } = false;
    
    [MaxLength(255)]
    [Column("address_en")]
    public string? AddressEn { get; set; }

    [MaxLength(255)]
    [Column("address_local")]
    public string? AddressLocal { get; set; }

    [ForeignKey(nameof(Province))]
    [Column("province_id")]
    public Guid? ProvinceId { get; set; }
    public DbProvince? Province { get; set; }

    [ForeignKey(nameof(District))]
    [Column("district_id")]
    public Guid? DistrictId { get; set; }
    public DbDistrict? District { get; set; }

    [ForeignKey(nameof(SubDistrict))]
    [Column("sub_district_id")]
    public Guid? SubDistrictId { get; set; }
    public DbSubDistrict? SubDistrict { get; set; }

    [MaxLength(20)]
    [Column("zip_code")]
    public string? ZipCode { get; set; }

    [MaxLength(320)]
    [Column("email")]
    public string? Email { get; set; }
    
    [MaxLength(20)]
    [Column("phone_number")]
    public string? PhoneNumber { get; set; }

    [MaxLength(20)]
    [Column("fax")]
    public string? Fax { get; set; }

    [Column("upload_group_id")]
    public Guid? UploadGroupId { get; set; }

    [Required]
    [Column("is_active")]
    public bool IsActive { get; set; } = false;

    public ICollection<SuUserBusiness>? UserBusinesses { get; set; }
    public ICollection<SuBusinessRole>? BusinessRoles { get; set; }
    public ICollection<SuBusinessAudit>? BusinessAudits { get; set; }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/su/SuBusiness.java
package com.softinter.sicapi.entity.su;

import com.softinter.sicapi.entity.base.BaseEntity;
import com.softinter.sicapi.entity.db.DbCountry;
import com.softinter.sicapi.entity.db.DbDistrict;
import com.softinter.sicapi.entity.db.DbProvince;
import com.softinter.sicapi.entity.db.DbSubDistrict;
import com.softinter.sicapi.entity.db.DbTitle;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "su_business")
public class SuBusiness extends BaseEntity {

    @Column(name = "tax_id", length = 30)
    private String taxId;

    @Column(name = "business_code", nullable = false, length = 30)
    private String businessCode;

    @Column(name = "branch_code", length = 30)
    private String branchCode;

    @Column(name = "person_type", nullable = false, length = 100)
    private String personType;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "title_id", nullable = false)
    private DbTitle title;

    @Column(name = "first_name_en", nullable = false, length = 100)
    private String firstNameEn;

    @Column(name = "middle_name_en", length = 100)
    private String middleNameEn;

    @Column(name = "last_name_en", length = 100)
    private String lastNameEn;

    @Column(name = "first_name_local", nullable = false, length = 100)
    private String firstNameLocal;

    @Column(name = "middle_name_local", length = 100)
    private String middleNameLocal;

    @Column(name = "last_name_local", length = 100)
    private String lastNameLocal;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "country_id", nullable = false)
    private DbCountry country;

    @Column(name = "support_local_address", nullable = false)
    private Boolean supportLocalAddress = false;

    @Column(name = "address_en", length = 255)
    private String addressEn;

    @Column(name = "address_local", length = 255)
    private String addressLocal;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "province_id")
    private DbProvince province;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "district_id")
    private DbDistrict district;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sub_district_id")
    private DbSubDistrict subDistrict;

    @Column(name = "zip_code", length = 20)
    private String zipCode;

    @Column(name = "email", length = 320)
    private String email;

    @Column(name = "phone_number", length = 20)
    private String phoneNumber;

    @Column(name = "fax", length = 20)
    private String fax;

    @Column(name = "upload_group_id")
    private UUID uploadGroupId;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = false;

    @OneToMany(mappedBy = "business", fetch = FetchType.LAZY)
    private List<SuUserBusiness> userBusinesses = new ArrayList<>();

    @OneToMany(mappedBy = "business", fetch = FetchType.LAZY)
    private List<SuBusinessRole> businessRoles = new ArrayList<>();

    @OneToMany(mappedBy = "business", fetch = FetchType.LAZY)
    private List<SuBusinessAudit> businessAudits = new ArrayList<>();
}
